frontend/src/metabase/hoc/ModalRoute.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from "react";
+import { Route } from "react-router";
+import { push } from "react-router-redux";
+import { connect } from "react-redux";
+import type { Location, LocationDescriptor } from "history";
+
+import Modal from "metabase/components/Modal";
+
+type IRoute = {
+  path: string;
+};
+
+export const getParentPath = (route: IRoute, location: Location) => {
+  const fullPathSegments = location.pathname.split("/");
+  const routeSegments = route.path.split("/");
+
+  fullPathSegments.splice(-routeSegments.length);
+
+  return fullPathSegments.join("/");
+};
+
+type ComposedModalProps = unknown & {
```

Comment:
why do we need `unknown` here? It just look unfamiliar to me, and removing it doesn't make TS complain about anything.